java-checks/src/main/java/org/sonar/java/checks/MethodOnlyCallsSuperCheck.java

```diff
@@ -156,4 +164,9 @@ private static boolean hasAnnotationDifferentFromOverride(List<AnnotationTree> a
     return false;
   }
 
+  private static boolean isClassAnnotatedWithTransactional(MethodTree methodTree) {
+    SymbolMetadata metadata = methodTree.symbol().enclosingClass().metadata();
```

Comment:
Let's double check we won't get NPE here

Reply:
`enclosingClass()` returns null for package symbol, since we are on a MethodTree, this is not possible.
I'm still in favor of defensive programming though, even if we will miss coverage, I will add it.